File: Assets/Bundles/Shared/Resources/TS/Util/TimeUtil.lua
-- Compiled with unity-ts v2.1.0-75
local TS = require("Shared/include/RuntimeLib")
local OnUpdate = require("Shared/TS/Util/Timer").OnUpdate
local startTime = os.time()
--[[
	*
	* Time synced between server and client.
]]
local function SharedTime()
	return InstanceFinder.TimeManager:TicksToTime(0)
end
local function WaitFrame()
	local promise = TS.Promise.new(function(resolve)
		OnUpdate:Once(function()
			resolve()
		end)
	end)
	promise:await()
end
local TimeUtil
do
	TimeUtil = setmetatable({}, {
		__tostring = function()
			return "TimeUtil"
		end,
	})
	TimeUtil.__index = TimeUtil
	function TimeUtil.new(...)
		local self = setmetatable({}, TimeUtil)
		return self:constructor(...) or self
	end
	function TimeUtil:constructor()
	end
	function TimeUtil:GetLifetimeSeconds()
		return os.time() - startTime
	end
	function TimeUtil:GetServerTime()
		return SharedTime()
	end
	function TimeUtil:GetDeltaTime()
		return Time.deltaTime
	end
	function TimeUtil:GetFixedDeltaTime()
		return Time.fixedDeltaTime
	end
end
return {
	SharedTime = SharedTime,
	WaitFrame = WaitFrame,
	TimeUtil = TimeUtil,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/Util/TimeUtil.ts
import { OnUpdate } from "./Timer";

let startTime = os.time();

/**
 * Time synced between server and client.
 */
export function SharedTime() {
	return InstanceFinder.TimeManager.TicksToTime(TickType.Tick);
}

export function WaitFrame() {
	const promise = new Promise<void>((resolve) => {
		OnUpdate.Once(() => {
			resolve();
		});
	});
	promise.await();
}

export class TimeUtil {
	/**
	 * @returns Time elapsed since server/client has started.
	 */
	public static GetLifetimeSeconds(): number {
		return os.time() - startTime;
	}

	/**
	 * @returns Time synchronized between server and client.
	 */
	public static GetServerTime(): number {
		return SharedTime();
	}

	/**
	 * @returns The interval in seconds from the last frame to the current one.
	 */
	public static GetDeltaTime(): number {
		return Time.deltaTime;
	}

	/**
	 * @returns The interval in seconds from the last physics frame to the current one.
	 */
	public static GetFixedDeltaTime(): number {
		return Time.fixedDeltaTime;
	}
}
